src/whylogs/core/statistics/constraints.py

```diff
@@ -437,7 +449,13 @@ def to_protobuf(self) -> ValueConstraintMsgs:
 
     def update(self, v):
         for c in self.constraints.values():
-            c.update(v)
+            if c.op not in (Op.MATCH, Op.NOMATCH):
```

Comment:
We will probably want to keep two separate lists of the contraints rather than check for the regex constraint as the switch for these methods, so there are raw value constraints and coerced typed constraints that can be added and we avoid the special logic for regex constraint in this code path and its bound at construction time instead?

Reply:
Yes, I wasn't sure if this was what you meant by having two types of updates. I'm going to update the PR with this change.